resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -393,7 +409,30 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
         val offerId = offer.getId.getValue
         val resources = remainingResources(offerId)
 
-        if (canLaunchTask(slaveId, resources)) {
+        var createTask = canLaunchTask(slaveId, resources)
+        if (hostToLocalTaskCount.nonEmpty) {
```

Comment:
The semantics you're going for is "Launch an executor on a host only when we have a task that wants to be on that host, or the configurable delay has elapsed", right?  By launching an executor on any host when we don't yet have any tasks to schedule, you're violating those semantics, right?